### ISSUE TITLE
Crash on 1.0.10

### ISSUE THREAD

**Issue by AntonioCS** [Jellybaby34/Dark-Souls-3-PvP-Watchdog#57]
Hey,

This is the 3rd time since installing this mod that I've had a crash
![image](https://user-images.githubusercontent.com/1086401/82736198-a79f3f80-9d1f-11ea-8fee-310cf519275e.png)
It says it's written to the logs but I don't really see anything on the Darks Souls 3 folder.
This crashes the game more or less an 1h into the game. I've been mostly doing invasions.

**Comment by AntonioCS** [comment]
Also, is this is on github where is the source code? Why is that not shared?

**Comment by LukeYui** [comment]
Hiya, thanks for the report. The crash info should be written under "YuiSpy" to "DS3PWLog.txt" in your DarkSoulsIII/Game folder (the same place you installed watchdog). 

The source code is not publicly available because it contains a lot of individual components which could be abused (e.g. anticheat bypass).

**Comment by AntonioCS** [comment]
Hey,
Here is screenshot of another crash, again after about 1h of game play (invasions only with Wex dust mostly):
![image](https://user-images.githubusercontent.com/1086401/82739845-2a81c380-9d3b-11ea-9c0f-2df2b2337604.png)

Right.. thought that file was just a startup log. Here is what is on the YuiSpy part:
```
[LOGERR 21:19:09] ----- [YuiSpy Activated] -----
[LOGERR 21:19:09] Exception Code: 0xC0000005 (ACCESS VIOLATION)
[LOGERR 21:19:09] Exception Flags: 0
[LOGERR 21:19:09] Fault Offset: 0x141769F02
[LOGERR 21:19:09] Number Parameters: 2
[LOGERR 21:19:09] The instruction at 0x141769F02 referenced memory at 0x0. The memory could not be written.
[LOGERR 21:19:09] Dumping session information:
[LOGERR 21:19:09] Generating register dump:
[LOGERR 21:19:09] RAX: 0x0
[LOGERR 21:19:09] RBX: 0x6D71258
[LOGERR 21:19:09] RCX: 0x0
[LOGERR 21:19:09] RDX: 0x0
[LOGERR 21:19:09] RDI: 0x1439E1788
[LOGERR 21:19:09] RSI: 0xD
[LOGERR 21:19:09] R8:  0x86CE758
[LOGERR 21:19:09] R9:  0x6D719C0
[LOGERR 21:19:09] R10: 0x0
[LOGERR 21:19:09] R11: 0x246
[LOGERR 21:19:09] R12: 0x7FF432467D90
[LOGERR 21:19:09] R13: 0x6D71DA8
[LOGERR 21:19:09] R14: 0x6D71E80
[LOGERR 21:19:09] R15: 0x7
[LOGERR 21:19:09] RBP: 0x6D719C0
[LOGERR 21:19:09] RSP: 0x86CE810
[LOGERR 21:19:09] ------ [YuiSpy Complete] -----
```
And here is the YuiSpy for the previous crash:
```
[LOGERR 18:02:00] ----- [YuiSpy Activated] -----
[LOGERR 18:02:00] Exception Code: 0xC0000005 (ACCESS VIOLATION)
[LOGERR 18:02:00] Exception Flags: 0
[LOGERR 18:02:00] Fault Offset: 0x141769F02
[LOGERR 18:02:00] Number Parameters: 2
[LOGERR 18:02:00] The instruction at 0x141769F02 referenced memory at 0x0. The memory could not be written.
[LOGERR 18:02:00] Dumping session information:
[LOGERR 18:02:00] Player Slot: 1, Name: MyDeprived, Steam ID: 011000010E5339AB, Owner Steam ID: NULL
[LOGERR 18:02:00] Player Slot: 2, Name: Maria Astoria, Steam ID: 0110000107180320, Owner Steam ID: 0110000107180320
[LOGERR 18:02:00] Player Slot: 3, Name: (-_-), Steam ID: 0110000102D8CC08, Owner Steam ID: 0110000102D8CC08
[LOGERR 18:02:00] Generating register dump:
[LOGERR 18:02:00] RAX: 0x0
[LOGERR 18:02:00] RBX: 0x7354DF8
[LOGERR 18:02:00] RCX: 0x0
[LOGERR 18:02:00] RDX: 0x0
[LOGERR 18:02:00] RDI: 0x1439E1788
[LOGERR 18:02:00] RSI: 0x3
[LOGERR 18:02:00] R8:  0x887E7C8
[LOGERR 18:02:00] R9:  0x7352140
[LOGERR 18:02:00] R10: 0x0
[LOGERR 18:02:00] R11: 0x246
[LOGERR 18:02:00] R12: 0x7FF4325D4DB0
[LOGERR 18:02:00] R13: 0x7354F60
[LOGERR 18:02:00] R14: 0x7354F70
[LOGERR 18:02:00] R15: 0x7
[LOGERR 18:02:00] RBP: 0x7352140
[LOGERR 18:02:00] RSP: 0x887E880
[LOGERR 18:02:00] ------ [YuiSpy Complete] -----
```

Is that a null pointer?

Regarding the source code. If someone wants to get into DS3 modding (new events and stuff like that) were would one go, since you can't show your code for "security reasons"

**Comment by LukeYui** [comment]
Thanks for posting the crash log! This is in an area of memory where dark souls crashes itself due to an unhandled exception. I believe it's to do with a memory error, which we've identified and fixed ready for patching in the next version.

In regards to DS3 modding, a good starting point may be the souls modding discord ?ServerName?, which is found here:  https://discord.gg/mT2JJjx

**Comment by AntonioCS** [comment]
So are you saying this is DS3 fault? I've never (or very rarely) had DS3 crash. Only after installing watchdog.
Also.. Is it normal for me to get a "connection error" when I press F1 to get rid of hackers while invading? I can open a new issue for this, but this one seems very strange. I've seen streamers use other "anti-cheat" mods that do kick players out while they are invading

**Comment by AntonioCS** [comment]
Thanks for sharing the discord link

**Comment by LukeYui** [comment]
It's not DS3's fault, it's a faulty patch in watchdog that's causing the memory error.

Yes that is normal. When invading, Watchdog disconnects you from the cheater. There are other anti-cheat programs out there, however they tend to patch common public cheats, so all it takes is for someone to tweak one of these cheats slightly and the protection is bypassed.

Watchdog on the other hand patches the underlying exploit, meaning that once something is patched it can no longer be abused by cheaters.

**Comment by AntonioCS** [comment]
Just had another crash after 1h44m of game play (invasions)
![image](https://user-images.githubusercontent.com/1086401/82742594-33cd5900-9d58-11ea-944c-d24ba4829c58.png)

```
[LOGERR 00:47:03] ----- [YuiSpy Activated] -----
[LOGERR 00:47:03] Exception Code: 0xC0000005 (ACCESS VIOLATION)
[LOGERR 00:47:03] Exception Flags: 0
[LOGERR 00:47:03] Fault Offset: 0x1401AFB70
[LOGERR 00:47:03] Number Parameters: 2
[LOGERR 00:47:03] The instruction at 0x1401AFB70 referenced memory at 0x8. The memory could not be written.
[LOGERR 00:47:03] Dumping session information:
[LOGERR 00:47:03] Player Slot: 1, Name: MyDeprived, Steam ID: 011000010E5339AB, Owner Steam ID: NULL
[LOGERR 00:47:03] Player Slot: 2, Name: PaiN, Steam ID: 0110000107887B3E, Owner Steam ID: 0110000107887B3E
[LOGERR 00:47:03] Generating register dump:
[LOGERR 00:47:03] RAX: 0x0
[LOGERR 00:47:03] RBX: 0x18
[LOGERR 00:47:03] RCX: 0xAF3DDE3A77080000
[LOGERR 00:47:03] RDX: 0x6E23170
[LOGERR 00:47:03] RDI: 0x0
[LOGERR 00:47:03] RSI: 0x0
[LOGERR 00:47:03] R8:  0x1428196A8
[LOGERR 00:47:03] R9:  0x0
[LOGERR 00:47:03] R10: 0x143D09A60
[LOGERR 00:47:03] R11: 0x4C33BF0
[LOGERR 00:47:03] R12: 0x6
[LOGERR 00:47:03] R13: 0x7FF432160200
[LOGERR 00:47:03] R14: 0x7FF431833750
[LOGERR 00:47:03] R15: 0x0
[LOGERR 00:47:03] RBP: 0x0
[LOGERR 00:47:03] RSP: 0x86FEBA0
[LOGERR 00:47:03] ------ [YuiSpy Complete] -----
```
If this is a known issue, please fix ASAP! Thanks!

**Comment by AntonioCS** [comment]
I guess watchdog didn't really want me to go against a known cheater
![image](https://user-images.githubusercontent.com/1086401/82753466-c5bb7d00-9dbd-11ea-963b-bbae64456cd9.png)
```
[LOGERR 12:54:08] ----- [YuiSpy Activated] -----
[LOGERR 12:54:08] Exception Code: 0xC0000005 (ACCESS VIOLATION)
[LOGERR 12:54:08] Exception Flags: 0
[LOGERR 12:54:08] Fault Offset: 0x14006483E
[LOGERR 12:54:08] Number Parameters: 2
[LOGERR 12:54:08] The instruction at 0x14006483E referenced memory at 0x0. The memory could not be read.
[LOGERR 12:54:08] Dumping session information:
[LOGERR 12:54:08] Player Slot: 1, Name: MyDeprived, Steam ID: 011000010E5339AB, Owner Steam ID: NULL
[LOGERR 12:54:08] Player Slot: 2, Name: Schattenaura, Steam ID: 0110000104F08099, Owner Steam ID: 0110000104F08099
[LOGERR 12:54:08] Player Slot: 3, Name: Rosa Luxemburg, Steam ID: 0110000141887A08, Owner Steam ID: 01100001078B1A5C
[LOGERR 12:54:08] Generating register dump:
[LOGERR 12:54:08] RAX: 0x0
[LOGERR 12:54:08] RBX: 0x0
[LOGERR 12:54:08] RCX: 0x0
[LOGERR 12:54:08] RDX: 0x0
[LOGERR 12:54:08] RDI: 0x7FF4381E2260
[LOGERR 12:54:08] RSI: 0x13A0ED60
[LOGERR 12:54:08] R8:  0x1447258B8
[LOGERR 12:54:08] R9:  0x1
[LOGERR 12:54:08] R10: 0x7FF42F895320
[LOGERR 12:54:08] R11: 0x0
[LOGERR 12:54:08] R12: 0x7FF42F19B340
[LOGERR 12:54:08] R13: 0x7FF431BEBAC0
[LOGERR 12:54:08] R14: 0x7FF433B4C6F0
[LOGERR 12:54:08] R15: 0x7FF431BEBA30
[LOGERR 12:54:08] RBP: 0x13A0EAD1
[LOGERR 12:54:08] RSP: 0x13A0EA10
[LOGERR 12:54:08] ------ [YuiSpy Complete] -----
```

**Comment by AntonioCS** [comment]
To note, I'll have to remove watchdog, this is making the game unplayable, the crash happened after another hour of game play

**Comment by AntonioCS** [comment]
And yet another crash:
![image](https://user-images.githubusercontent.com/1086401/82764370-15be3200-9e06-11ea-818a-0b4a62b0374f.png)

```
[LOGERR 21:31:33] ----- [YuiSpy Activated] -----
[LOGERR 21:31:34] Exception Code: 0xC0000005 (ACCESS VIOLATION)
[LOGERR 21:31:34] Exception Flags: 0
[LOGERR 21:31:34] Fault Offset: 0x141769F02
[LOGERR 21:31:34] Number Parameters: 2
[LOGERR 21:31:34] The instruction at 0x141769F02 referenced memory at 0x0. The memory could not be written.
[LOGERR 21:31:34] Dumping session information:
[LOGERR 21:31:34] Player Slot: 1, Name: MyDeprived, Steam ID: 011000010E5339AB, Owner Steam ID: NULL
[LOGERR 21:31:34] Player Slot: 2, Name: Vektor, Steam ID: 011000010BBE4C05, Owner Steam ID: 011000010BBE4C05
[LOGERR 21:31:34] Generating register dump:
[LOGERR 21:31:34] RAX: 0x0
[LOGERR 21:31:34] RBX: 0x72F5398
[LOGERR 21:31:34] RCX: 0x0
[LOGERR 21:31:34] RDX: 0x0
[LOGERR 21:31:34] RDI: 0x1439E1788
[LOGERR 21:31:34] RSI: 0xD
[LOGERR 21:31:34] R8:  0x871E748
[LOGERR 21:31:34] R9:  0x72F5960
[LOGERR 21:31:34] R10: 0x0
[LOGERR 21:31:34] R11: 0x246
[LOGERR 21:31:34] R12: 0x7FF49E3F04D8
[LOGERR 21:31:34] R13: 0x72F5D48
[LOGERR 21:31:34] R14: 0x72F5E20
[LOGERR 21:31:34] R15: 0x7
[LOGERR 21:31:34] RBP: 0x72F5960
[LOGERR 21:31:34] RSP: 0x871E800
[LOGERR 21:31:34] ------ [YuiSpy Complete] -----
```

Yeah... going to remove this now

**Comment by LukeYui** [comment]
Sorry to hear you've been having problems with the mod. I believe all of the crashes you're reporting here are related to that 1 faulty patch in watchdog right now. An update should be out soon which should resolve these issues.

**Comment by AntonioCS** [comment]
@LukeYui yeah, I get it and the mod does seem to warn me of hackers, how many people are in the game which is quite useful.. but the crashes are killing me...
If this is a known issue why isn't this flagged as critical? I've posted multiple crashes and it seems I am not alone: https://github.com/Jellybaby34/Dark-Souls-3-PvP-Watchdog/issues/56
Is there a timeline for a fix?

**Comment by LukeYui** [comment]
All crashes are urgent priority. Every change we make to the code also goes through a 2 week ban testing period before we make it public. I understand it's frustrating for users and it is for us too, but we'd much rather be over cautious with updates and wait longer between them than risk any bans.

We'll probably have the next version out after the next ban wave has passed (on Thursday next week).

**Comment by AntonioCS** [comment]
I can't believe I am the only one having these crashes:
![image](https://user-images.githubusercontent.com/1086401/82832783-0a294480-9eb4-11ea-9f16-b958ab91abc3.png)

Normally the crash happens at the start of an invasion this was 5m in (after about 1h of gameplay)
```
[LOGERR 18:17:05] ----- [YuiSpy Activated] -----
[LOGERR 18:17:05] Exception Code: 0xC0000005 (ACCESS VIOLATION)
[LOGERR 18:17:05] Exception Flags: 0
[LOGERR 18:17:05] Fault Offset: 0x14006483E
[LOGERR 18:17:05] Number Parameters: 2
[LOGERR 18:17:05] The instruction at 0x14006483E referenced memory at 0x0. The memory could not be read.
[LOGERR 18:17:05] Dumping session information:
[LOGERR 18:17:05] Player Slot: 1, Name: MyDeprived, Steam ID: 011000010E5339AB, Owner Steam ID: NULL
[LOGERR 18:17:05] Player Slot: 2, Name: Fall, Steam ID: 011000013D504861, Owner Steam ID: 011000013D504861
[LOGERR 18:17:05] Player Slot: 3, Name: Violeto-Chan, Steam ID: 01100001065E638D, Owner Steam ID: 01100001065E638D
[LOGERR 18:17:05] Generating register dump:
[LOGERR 18:17:05] RAX: 0x0
[LOGERR 18:17:05] RBX: 0x0
[LOGERR 18:17:05] RCX: 0x0
[LOGERR 18:17:05] RDX: 0x0
[LOGERR 18:17:05] RDI: 0x7FF43314E760
[LOGERR 18:17:05] RSI: 0x1390ED60
[LOGERR 18:17:05] R8:  0x1447258B8
[LOGERR 18:17:05] R9:  0x2
[LOGERR 18:17:05] R10: 0x7FF436F41980
[LOGERR 18:17:05] R11: 0x0
[LOGERR 18:17:05] R12: 0x7FF42F1854A0
[LOGERR 18:17:05] R13: 0x7FF436F41290
[LOGERR 18:17:05] R14: 0x7FF4381AC900
[LOGERR 18:17:05] R15: 0x7FF43314DE80
[LOGERR 18:17:05] RBP: 0x1390EAD1
[LOGERR 18:17:05] RSP: 0x1390EA10
[LOGERR 18:17:05] ------ [YuiSpy Complete] -----
```

**Comment by LukeYui** [comment]
I think a fair few people have crashes similar. This crash is related to that 1 faulty patch again.

It's quite tricky to test and debug because I cannot recreate any of these issues myself (watchdog runs fine for me, I've used it for 5 or 6 hours continuously with no problems) but we have an idea of what could be causing it for other people, and others are understandably frustrated that watchdog is being very crash happy for them.

**Comment by AntonioCS** [comment]
Another crash after 1h30m of gameplay 
![image](https://user-images.githubusercontent.com/1086401/82930161-d82fe500-9f7c-11ea-9c47-8acd9fec7134.png)

```
[LOGERR 18:14:23] ----- [YuiSpy Activated] -----
[LOGERR 18:14:23] Exception Code: 0xC0000005 (ACCESS VIOLATION)
[LOGERR 18:14:23] Exception Flags: 0
[LOGERR 18:14:23] Fault Offset: 0x141769F02
[LOGERR 18:14:23] Number Parameters: 2
[LOGERR 18:14:23] The instruction at 0x141769F02 referenced memory at 0x0. The memory could not be written.
[LOGERR 18:14:23] Dumping session information:
[LOGERR 18:14:23] Player Slot: 1, Name: MyDeprived, Steam ID: 011000010E5339AB, Owner Steam ID: NULL
[LOGERR 18:14:23] Player Slot: 2, Name: Triss, Steam ID: 011000011A433F81, Owner Steam ID: 011000011A433F81
[LOGERR 18:14:23] Player Slot: 3, Name: zuita, Steam ID: 011000010B4CA755, Owner Steam ID: 011000010B4CA755
[LOGERR 18:14:23] Player Slot: 4, Name: Ciri, Steam ID: 0110000109082ADE, Owner Steam ID: 0110000109082ADE
[LOGERR 18:14:23] Generating register dump:
[LOGERR 18:14:23] RAX: 0x0
[LOGERR 18:14:23] RBX: 0x6D53528
[LOGERR 18:14:23] RCX: 0x0
[LOGERR 18:14:23] RDX: 0x0
[LOGERR 18:14:23] RDI: 0x1439E1788
[LOGERR 18:14:23] RSI: 0x1C
[LOGERR 18:14:23] R8:  0x86BE7C8
[LOGERR 18:14:23] R9:  0x6D536E0
[LOGERR 18:14:23] R10: 0x0
[LOGERR 18:14:23] R11: 0x246
[LOGERR 18:14:24] R12: 0x7FF49E2C4998
[LOGERR 18:14:24] R13: 0x6D54928
[LOGERR 18:14:24] R14: 0x6D54AD0
[LOGERR 18:14:24] R15: 0x7
[LOGERR 18:14:24] RBP: 0x6D536E0
[LOGERR 18:14:24] RSP: 0x86BE880
[LOGERR 18:14:24] ------ [YuiSpy Complete] -----
```

**Comment by AntonioCS** [comment]
One more for the pile:
![image](https://user-images.githubusercontent.com/1086401/83071754-87e08200-a065-11ea-8d23-3cbc3bca1643.png)

```
[LOGERR 22:00:11] ----- [YuiSpy Activated] -----
[LOGERR 22:00:11] Exception Code: 0xC0000005 (ACCESS VIOLATION)
[LOGERR 22:00:11] Exception Flags: 0
[LOGERR 22:00:11] Fault Offset: 0x1401AF73D
[LOGERR 22:00:11] Number Parameters: 2
[LOGERR 22:00:11] The instruction at 0x1401AF73D referenced memory at 0x518. The memory could not be read.
[LOGERR 22:00:11] Dumping session information:
[LOGERR 22:00:11] Player Slot: 1, Name: MyDeprived, Steam ID: 011000010E5339AB, Owner Steam ID: NULL
[LOGERR 22:00:11] Player Slot: 2, Name: Gen, Steam ID: 01100001087A734E, Owner Steam ID: 01100001087A734E
[LOGERR 22:00:11] Generating register dump:
[LOGERR 22:00:11] RAX: 0x3BF7E8
[LOGERR 22:00:11] RBX: 0x0
[LOGERR 22:00:11] RCX: 0x3BF7E888
[LOGERR 22:00:11] RDX: 0x88
[LOGERR 22:00:11] RDI: 0xE358FCFA
[LOGERR 22:00:11] RSI: 0x68DE0D4
[LOGERR 22:00:11] R8:  0x68DE1D0
[LOGERR 22:00:11] R9:  0x68DE1D0
[LOGERR 22:00:11] R10: 0x508
[LOGERR 22:00:11] R11: 0x144714CF0
[LOGERR 22:00:11] R12: 0x7FF49E35DD48
[LOGERR 22:00:11] R13: 0x0
[LOGERR 22:00:11] R14: 0xFC
[LOGERR 22:00:11] R15: 0x9
[LOGERR 22:00:11] RBP: 0x68DE0D4
[LOGERR 22:00:11] RSP: 0x865E9C0
[LOGERR 22:00:11] ------ [YuiSpy Complete] -----
```

**Comment by LukeYui** [comment]
The new release is out. It *should* address the crashes you've been having, but please do let me know if it hasn't. Thanks again

**Comment by AntonioCS** [comment]
@LukeYui I'll test it today! Thanks!!!

**Comment by AntonioCS** [comment]
Just downloaded the dll from the master branch and if this is a new version, shouldn't the version number be changed to something else? Still seeing 1.0.10

**Comment by AntonioCS** [comment]
Well... unless I got the wrong dll, the fix didn't work:
![image](https://user-images.githubusercontent.com/1086401/83242731-c45ecb80-a194-11ea-89cd-b4f80b07128a.png)

```
[LOGERR 10:10:50] ----- [YuiSpy Activated] -----
[LOGERR 10:10:50] Exception Code: 0xC0000005 (ACCESS VIOLATION)
[LOGERR 10:10:50] Exception Flags: 0
[LOGERR 10:10:50] Fault Offset: 0x141769F02
[LOGERR 10:10:50] Number Parameters: 2
[LOGERR 10:10:50] The instruction at 0x141769F02 referenced memory at 0x0. The memory could not be written.
[LOGERR 10:10:50] Dumping session information:
[LOGERR 10:10:50] Generating register dump:
[LOGERR 10:10:50] RAX: 0x0
[LOGERR 10:10:50] RBX: 0x728AA98
[LOGERR 10:10:50] RCX: 0x0
[LOGERR 10:10:50] RDX: 0x0
[LOGERR 10:10:50] RDI: 0x1439E1788
[LOGERR 10:10:50] RSI: 0x1C
[LOGERR 10:10:50] R8:  0x86AE7C8
[LOGERR 10:10:50] R9:  0x728AE10
[LOGERR 10:10:50] R10: 0x0
[LOGERR 10:10:50] R11: 0x246
[LOGERR 10:10:50] R12: 0x7FF49E2D0E08
[LOGERR 10:10:50] R13: 0x728BD88
[LOGERR 10:10:50] R14: 0x728BE30
[LOGERR 10:10:50] R15: 0x5
[LOGERR 10:10:50] RBP: 0x728AE10
[LOGERR 10:10:50] RSP: 0x86AE880
[LOGERR 10:10:50] ------ [YuiSpy Complete] -----
```
This was again more or less after 1h of gameplay

**Comment by LukeYui** [comment]
Ah, no that's the correct version. Apologies for this, I'm going to look into this more and try to get a fix out ASAP.

**Comment by AntonioCS** [comment]
Well.. if you release an update and the version stays the same.... it's kinda weird.
Maybe 1.0.10.1 or 1.0.11, something... Something that will tell me that I have a new version...
That or fix the bug, that will definitely tell me I am using a new version :)

**Comment by AntonioCS** [comment]
Another crash:
![image](https://user-images.githubusercontent.com/1086401/83268045-f8e67d80-a1bc-11ea-9078-8e338ffe4160.png)
The crash seems to happen always around 1h and a bit of gameplay.
```
[LOGERR 14:58:36] ----- [YuiSpy Activated] -----
[LOGERR 14:58:36] Exception Code: 0xC0000005 (ACCESS VIOLATION)
[LOGERR 14:58:36] Exception Flags: 0
[LOGERR 14:58:36] Fault Offset: 0x141769F02
[LOGERR 14:58:36] Number Parameters: 2
[LOGERR 14:58:36] The instruction at 0x141769F02 referenced memory at 0x0. The memory could not be written.
[LOGERR 14:58:36] Dumping session information:
[LOGERR 14:58:36] Generating register dump:
[LOGERR 14:58:36] RAX: 0x0
[LOGERR 14:58:36] RBX: 0x6FB8DB8
[LOGERR 14:58:36] RCX: 0x0
[LOGERR 14:58:36] RDX: 0x0
[LOGERR 14:58:36] RDI: 0x1439E1788
[LOGERR 14:58:36] RSI: 0x4
[LOGERR 14:58:36] R8:  0x88AE748
[LOGERR 14:58:36] R9:  0x6FB7680
[LOGERR 14:58:36] R10: 0x0
[LOGERR 14:58:36] R11: 0x246
[LOGERR 14:58:36] R12: 0x7FF49E2C4698
[LOGERR 14:58:36] R13: 0x6FB82B8
[LOGERR 14:58:36] R14: 0x6F698F0
[LOGERR 14:58:36] R15: 0x7
[LOGERR 14:58:36] RBP: 0x6FB7680
[LOGERR 14:58:36] RSP: 0x88AE800
[LOGERR 14:58:36] ------ [YuiSpy Complete] -----
```

**Comment by LukeYui** [comment]
From the screenshots you've posted, the vast majority seem to be happening on load screens. Is this just as you're loading in to a new session?

I'm also curious on whether you'd be able to help with the debugging process some more, as I cannot recreate this crashes myself - If you feel up to it could you help me find out exactly where the fault is occurring? If you don't want to or don't have time though, no worries!

**Comment by AntonioCS** [comment]
Another crash and yes... seems to be on the majority (not all) on the loading screens.
![image](https://user-images.githubusercontent.com/1086401/83288407-3528d680-a1db-11ea-929c-98482f75de29.png)
This time it was at 2h+ gameplay
```
[LOGERR 18:35:06] ----- [YuiSpy Activated] -----
[LOGERR 18:35:06] Exception Code: 0xC0000005 (ACCESS VIOLATION)
[LOGERR 18:35:06] Exception Flags: 0
[LOGERR 18:35:06] Fault Offset: 0x1401AF73D
[LOGERR 18:35:06] Number Parameters: 2
[LOGERR 18:35:06] The instruction at 0x1401AF73D referenced memory at 0x518. The memory could not be read.
[LOGERR 18:35:06] Dumping session information:
[LOGERR 18:35:06] Player Slot: 1, Name: MyDeprived, Steam ID: 011000010E5339AB, Owner Steam ID: NULL
[LOGERR 18:35:06] Player Slot: 2, Name: Ator, Steam ID: 0110000103DE5F8E, Owner Steam ID: 0110000103DE5F8E
[LOGERR 18:35:06] Generating register dump:
[LOGERR 18:35:06] RAX: 0x54BE48
[LOGERR 18:35:06] RBX: 0x0
[LOGERR 18:35:06] RCX: 0x54BE48F5
[LOGERR 18:35:06] RDX: 0xF5
[LOGERR 18:35:06] RDI: 0xCD83B8DB
[LOGERR 18:35:06] RSI: 0x7FF430814000
[LOGERR 18:35:06] R8:  0x7FF430814070
[LOGERR 18:35:06] R9:  0x7FF430814070
[LOGERR 18:35:06] R10: 0x508
[LOGERR 18:35:06] R11: 0x144714CF0
[LOGERR 18:35:06] R12: 0x7FF4324F7440
[LOGERR 18:35:06] R13: 0x0
[LOGERR 18:35:06] R14: 0x70
[LOGERR 18:35:06] R15: 0x4
[LOGERR 18:35:06] RBP: 0x7FF430814000
[LOGERR 18:35:06] RSP: 0x878E9C0
[LOGERR 18:35:06] ------ [YuiSpy Complete] -----
```

**Comment by AntonioCS** [comment]
@LukeYui Yes, I would like to help as much as possible

**Comment by AntonioCS** [comment]
@LukeYui you can see from my multiple crashes, that I am playing the game multiple times a day... so... for now my schedule is kinda open :D

**Comment by LukeYui** [comment]
Thank you! First off, are you using absolutely any mods at all that you know of? Even other gaming applications like RivaTuner/MSI afterburner with DS3?

**Comment by AntonioCS** [comment]
The only other mod is wex dust. I don't think I have any of those applications. I have updated my graphics card driver (today actually)

**Comment by AntonioCS** [comment]
Just played for almost 3h with not crashes :D Maybe all I need was an update to my graphics card. Not sure how it's related because the game never crashed before I installed the mod... but hey.. as long it works I'm happy.

I'll keep you updated if this changes.

**Comment by LukeYui** [comment]
Apologies for the delayed response. Wow that's a surprise, perhaps it was something with that graphics driver then?

**Comment by AntonioCS** [comment]
Hello,
Well... sorry to say but no... there is something else going one, just had another crash:
![image](https://user-images.githubusercontent.com/1086401/83337481-a6d45380-a2b3-11ea-87dd-ff3df3c0f977.png)

```
[LOGERR 20:23:55] ----- [YuiSpy Activated] -----
[LOGERR 20:23:55] Exception Code: 0xC0000005 (ACCESS VIOLATION)
[LOGERR 20:23:55] Exception Flags: 0
[LOGERR 20:23:55] Fault Offset: 0x1401AF73D
[LOGERR 20:23:55] Number Parameters: 2
[LOGERR 20:23:55] The instruction at 0x1401AF73D referenced memory at 0x518. The memory could not be read.
[LOGERR 20:23:55] Dumping session information:
[LOGERR 20:23:55] Player Slot: 1, Name: MyDeprived, Steam ID: 011000010E5339AB, Owner Steam ID: NULL
[LOGERR 20:23:55] Player Slot: 2, Name: Mikita, Steam ID: 011000010D6598B3, Owner Steam ID: 011000010D6598B3
[LOGERR 20:23:55] Player Slot: 3, Name: Petya, Steam ID: 011000011A590076, Owner Steam ID: 011000011A590076
[LOGERR 20:23:55] Generating register dump:
[LOGERR 20:23:55] RAX: 0x54BE48
[LOGERR 20:23:55] RBX: 0x0
[LOGERR 20:23:55] RCX: 0x54BE48F5
[LOGERR 20:23:55] RDX: 0xF5
[LOGERR 20:23:55] RDI: 0xCD83B8DB
[LOGERR 20:23:55] RSI: 0x7FF4325D65A0
[LOGERR 20:23:55] R8:  0x7FF4325D6610
[LOGERR 20:23:55] R9:  0x7FF4325D6610
[LOGERR 20:23:55] R10: 0x508
[LOGERR 20:23:55] R11: 0x144714CF0
[LOGERR 20:23:55] R12: 0x7FF4324F7440
[LOGERR 20:23:55] R13: 0x0
[LOGERR 20:23:55] R14: 0x70
[LOGERR 20:23:55] R15: 0x4
[LOGERR 20:23:55] RBP: 0x7FF4325D65A0
[LOGERR 20:23:55] RSP: 0x871E9C0
[LOGERR 20:23:55] ------ [YuiSpy Complete] -----
```

**Comment by LukeYui** [comment]
Ah sorry to hear this, I though that might have been it. We'll push an update to the beta branch shortly with a more detailed crash report logger, that way we might be able to see what part of memory the game is failing on.

**Comment by AntonioCS** [comment]
Cool, let me know when to get it.
Also, I've joined the discord server for mods

**Comment by AntonioCS** [comment]
Sundays crash log after 1h40m game play
![image](https://user-images.githubusercontent.com/1086401/83358546-31788980-a36c-11ea-9515-d9b0c8aa6c6f.png)
```
[LOGERR 18:24:57] ----- [YuiSpy Activated] -----
[LOGERR 18:24:57] Exception Code: 0xC0000005 (ACCESS VIOLATION)
[LOGERR 18:24:57] Exception Flags: 0
[LOGERR 18:24:57] Fault Offset: 0x141769F02
[LOGERR 18:24:57] Number Parameters: 2
[LOGERR 18:24:57] The instruction at 0x141769F02 referenced memory at 0x0. The memory could not be written.
[LOGERR 18:24:57] Dumping session information:
[LOGERR 18:24:57] Player Slot: 1, Name: MyDeprived, Steam ID: 011000010E5339AB, Owner Steam ID: NULL
[LOGERR 18:24:57] Player Slot: 2, Name: indeferentno, Steam ID: 011000010A46F406, Owner Steam ID: 011000010A46F406
[LOGERR 18:24:57] Generating register dump:
[LOGERR 18:24:57] RAX: 0x0
[LOGERR 18:24:57] RBX: 0x6F2C958
[LOGERR 18:24:57] RCX: 0x0
[LOGERR 18:24:57] RDX: 0x0
[LOGERR 18:24:57] RDI: 0x1439E1788
[LOGERR 18:24:57] RSI: 0x9
[LOGERR 18:24:57] R8:  0x88AE7C8
[LOGERR 18:24:57] R9:  0x6F2CF90
[LOGERR 18:24:57] R10: 0x0
[LOGERR 18:24:57] R11: 0x246
[LOGERR 18:24:57] R12: 0x7FF49E4D58C8
[LOGERR 18:24:57] R13: 0x6F2CF60
[LOGERR 18:24:57] R14: 0x6F0F4E0
[LOGERR 18:24:57] R15: 0x7
[LOGERR 18:24:57] RBP: 0x6F2CF90
[LOGERR 18:24:57] RSP: 0x88AE880
[LOGERR 18:24:57] ------ [YuiSpy Complete] -----
```

Earlier today I played for around 1h without issues.

**Comment by LukeYui** [comment]
It's hard to know about this crash, I did some digging the other day and am aware that connecting to some versions of the Cinders mod causes this issue. 

It seems that hosts with Cinders send a massive amount of event packets which overwhelms the memory Dark Souls puts aside for storing and keeping track of received data. Once that's gone it causes this crash. I'm still looking into Watchdog being the issue because you have very consistent crashes after a certain amount of time in-game, however it's difficult to know how much is Watchdog, and how much is connecting to modded games.

I'm adding something to Watchdog that will keep a close eye on session memory, which should stop Cinders users crashing connecting clients, and also prevent malicious cheaters sending you too many packets at once. We're also looking into increasing the amount of memory available to sessions to make these crashes more difficult, meaning the client sending them will most likely run out of memory before Watchdog does, and you will automatically disconnect from Cinder's hosts.

**Comment by AntonioCS** [comment]
Another crash (again on connecting to a session for invasion).
Regarding what you mentioned regarding "received data", does this have any connection to it: https://github.com/Jellybaby34/Dark-Souls-3-PvP-Watchdog/issues/61
First time I've seen a spam on the log regarding player id.


Crash data:

![image](https://user-images.githubusercontent.com/1086401/83361853-609af500-a384-11ea-8c26-68ef0b5326fa.png)
```
[LOGERR 21:17:48] ----- [YuiSpy Activated] -----
[LOGERR 21:17:48] Exception Code: 0xC0000005 (ACCESS VIOLATION)
[LOGERR 21:17:48] Exception Flags: 0
[LOGERR 21:17:48] Fault Offset: 0x141769F02
[LOGERR 21:17:48] Number Parameters: 2
[LOGERR 21:17:48] The instruction at 0x141769F02 referenced memory at 0x0. The memory could not be written.
[LOGERR 21:17:48] Dumping session information:
[LOGERR 21:17:48] Player Slot: 1, Name: MyDeprived, Steam ID: 011000010E5339AB, Owner Steam ID: NULL
[LOGERR 21:17:48] Player Slot: 2, Name: Nieselpriem, Steam ID: 01100001008563D7, Owner Steam ID: 01100001008563D7
[LOGERR 21:17:48] Player Slot: 3, Name: Orenn, Steam ID: 0110000117145EAC, Owner Steam ID: 0110000117145EAC
[LOGERR 21:17:48] Generating register dump:
[LOGERR 21:17:48] RAX: 0x0
[LOGERR 21:17:48] RBX: 0x7395388
[LOGERR 21:17:48] RCX: 0x0
[LOGERR 21:17:48] RDX: 0x0
[LOGERR 21:17:48] RDI: 0x1439E1788
[LOGERR 21:17:48] RSI: 0xD
[LOGERR 21:17:48] R8:  0x88FE7C8
[LOGERR 21:17:48] R9:  0x7395A70
[LOGERR 21:17:48] R10: 0x0
[LOGERR 21:17:48] R11: 0x246
[LOGERR 21:17:48] R12: 0x7FF49E3F0218
[LOGERR 21:17:48] R13: 0x7395E38
[LOGERR 21:17:48] R14: 0x7395EF0
[LOGERR 21:17:48] R15: 0x6
[LOGERR 21:17:48] RBP: 0x7395A70
[LOGERR 21:17:48] RSP: 0x88FE880
[LOGERR 21:17:48] ------ [YuiSpy Complete] -----
```

**Comment by LukeYui** [comment]
Is there any way I can contact you through a private DM? I feel like it would make this a bit easier

**Comment by AntonioCS** [comment]
@LukeYui yes! As mentioned I've joined the ?ServerName? discord. Are you not there?
Not sure how to search for your nickname... kinda new to discord... guess nobody uses IRC these days :D

**Comment by LukeYui** [comment]
I sent you a friend invite (I think).. you were inquiring about WEX dust on the modding server? If not could you send me your username, I might have added someone random

**Comment by AntonioCS** [comment]
Yes... it was me asking about wex dust and being ignored :D
